SignalServiceKit/src/Messages/Interactions/TSOutgoingMessage.m

```diff
@@ -498,6 +406,44 @@ - (OWSSignalServiceProtosDataMessage *)buildDataMessage:(NSString *_Nullable)rec
     OWSSignalServiceProtosDataMessageBuilder *builder = [self dataMessageBuilder];
     [builder setTimestamp:self.timestamp];
     [builder addLocalProfileKeyIfNecessary:self.thread recipientId:recipientId];
+
+    if (self.quotedMessage) {
+        OWSSignalServiceProtosDataMessageQuoteBuilder *quoteBuilder =
+            [OWSSignalServiceProtosDataMessageQuoteBuilder new];
+        [quoteBuilder setId:self.quotedMessage.timestamp];
+        [quoteBuilder setAuthor:self.quotedMessage.authorId];
+
+        BOOL hasQuotedText = NO;
+        BOOL hasQuotedAttachment = NO;
+        if (self.quotedMessage.body.length > 0) {
+            [quoteBuilder setText:self.quotedMessage.body];
+
+            hasQuotedText = YES;
+        }
+
+        if (self.quotedMessage.contentType.length > 0) {
+
+            OWSSignalServiceProtosAttachmentPointerBuilder *attachmentBuilder =
+                [OWSSignalServiceProtosAttachmentPointerBuilder new];
+            if (self.quotedMessage.thumbnailData.length > 0) {
+                [attachmentBuilder setThumbnail:self.quotedMessage.thumbnailData];
```

Comment:
Moxie requested that we don't send thumbnail data this way for now, and instead upload the thumbnail file.

Reply:
Per offline discussion - we can leave this as is for the purposes of this PR and address it subsequently.

Reply:
+1